### PR TITLE
Add support for macOS(Darwin) / Windows

### PR DESCRIPTION
**Pull request by CoreNion**
References: #22 

Support for macOS was achieved by leveraging `sharedDarwinSource` and diverting the iOS source.

### CODE REVIEW

**Comment by chipweinberger** [comment]
yes please!

**Comment by jsiedentop** [comment]
Do i understand correctly that we are currently only waiting for a review by a verified googler like @jonasfj? I would be really grateful if someone could approve this pull request. Thank you very much!

File: src/webcrypto.h
Comment (jonasfj) [inline src/webcrypto.h:29]: Missing newline

File: windows/CMakeLists.txt
Comment (jonasfj) [inline windows/CMakeLists.txt:1]: Missing license header

**Review by jonasfj — APPROVED** [review]
r+

**Comment by jonasfj** [comment]
For future reference please do consider making smaller PRs, and feel free to ping me or hit me up on discord. There is always too many things going on at the same time to keep track of everything, so stuff slips through the cracks.

**Comment by jonasfj** [comment]
@jsiedentop I think we should get this published, but I'm a bit concerned about releasing it without any tests. It just looks too good to be sitting around.

Please feel encouraged to contribute on https://github.com/google/webcrypto.dart/issues/76, I'm not on macos and debugging why the integration tests are hanging is not easy using Github Actions :D

**Pull request merged**
